src/wallet/wallet.ts

```diff
@@ -1498,15 +1500,16 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
       }
     }
 
-    const isNFT = newOptions.nftData !== null;
+    const isNFT =
+      newOptions.isCreateNFT || (newOptions.data !== null && newOptions.data.length > 0);
 
     const depositPercent = this.storage.getTokenDepositPercentage();
     // 1. Calculate HTR deposit needed
     let deposit = tokens.getDepositAmount(amount, depositPercent);
 
-    if (isNFT) {
-      // For NFT we have a fee of 0.01 HTR, then the deposit utxo query must get an additional 1
-      deposit += 1n;
+    if (isNFT && newOptions.data && newOptions.data.length > 0) {
```

Comment:
suggestion: The definition of `isNFT` above already includes `newOptions.data?.length > 0`. I guess we can remove this redundant validation here.